### PR TITLE
Add Azure PCI DSS 3.2.1 (Built-in Regulatory Compliance ) Closes #142

### LINKED ISSUE #142: Add Azure PCI DSS 3.2.1 (Built-in Regulatory Compliance )
**Is your feature request related to a problem? Please describe.**

[Reference](https://learn.microsoft.com/en-us/azure/governance/policy/samples/pci-dss-3-2-1)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Azure Compliance Mod for Steampipe
 
-200+ checks covering industry defined security best practices for Azure. Includes full support for `CIS v1.3`,`CIS v1.4`,`CIS v1.5`, `HIPAA HITRUST 9.2` and `NIST SP 800-53` compliance benchmarks across all your Azure subscriptions.
+200+ checks covering industry defined security best practices for Azure. Includes full support for `CIS v1.3`,`CIS v1.4`,`CIS v1.5`, `HIPAA HITRUST 9.2`, `NIST SP 800-53` and `PCI DSS v3.2.1` compliance benchmarks across all your Azure subscriptions.
```
Comment (madhushreeray30) [inline README.md:3]: - [ ] 
```suggestion
200+ checks covering industry defined security best practices for Azure. Includes full support for `CIS v1.3`, `CIS v1.4`, `CIS v1.5`, `HIPAA HITRUST 9.2`, `NIST SP 800-53` and `PCI DSS v3.2.1` compliance benchmarks across all your Azure subscriptions.
```

File: pci_dss_v321/pci_dss.sp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+locals {
+  pci_dss_v321_common_tags = merge(local.azure_compliance_common_tags, {
+    pci_dss_v321 = "true"
+    type         = "Benchmark"
+  })
+}
+
+benchmark "pci_dss_v321" {
+  title       = "PCI DSS 3.2.1"
```
Comment (madhushreeray30) [inline pci_dss_v321/pci_dss.sp:9]:
```suggestion
  title         = "PCI DSS 3.2.1"
```

File: pci_dss_v321/pci_dss.sp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  description = "The PCI DSS applies to all entities that store, process, and/or transmit cardholder data. It covers technical and operational system components included in or connected to cardholder data. If you accept or process payment cards, PCI DSS applies to you."
```
Comment (madhushreeray30) [inline pci_dss_v321/pci_dss.sp:10]:
```suggestion
  description   = "The PCI DSS applies to all entities that store, process, and/or transmit cardholder data. It covers technical and operational system components included in or connected to cardholder data. If you accept or process payment cards, PCI DSS applies to you."
```

File: pci_dss_v321/pci_dss.sp [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  documentation = file("./pci_dss_v321/docs/pci_dss_v321_overview.md")
```
Comment (madhushreeray30) [inline pci_dss_v321/pci_dss.sp:11]: benchmark "pci_dss_v321" {
  title         = "PCI DSS 3.2.1"
  description   = "The PCI DSS applies to all entities that store, process, and/or transmit cardholder data. It covers technical and operational system components included in or connected to cardholder data. If you accept or process payment cards, PCI DSS applies to you."
  documentation = file("./pci_dss_v321/docs/pci_dss_v321_overview.md")
```suggestion
benchmark "pci_dss_v321" {
  title         = "PCI DSS 3.2.1"
  description   = "The PCI DSS applies to all entities that store, process, and/or transmit cardholder data. It covers technical and operational system components included in or connected to cardholder data. If you accept or process payment cards, PCI DSS applies to you."
  documentation = file("./pci_dss_v321/docs/pci_dss_v321_overview.md")
```

File: pci_dss_v321/requirement_8.sp [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    benchmark.pci_dss_v321_requirement_8_1,
+    benchmark.pci_dss_v321_requirement_8_2,
+    benchmark.pci_dss_v321_requirement_8_3
+  ]
+
+  tags = local.pci_dss_v321_common_tags
+}
+
+benchmark "pci_dss_v321_requirement_8_1" {
+  title       = "Define and implement policies and procedures to ensure proper user identification management for users and administrators on all system components. Assign all users a unique user name before allowing them to access system components or cardholder data"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:14]:
```suggestion
benchmark "pci_dss_v321_requirement_8_1" {
  title = "Define and implement policies and procedures to ensure proper user identification management for users and administrators on all system components. Assign all users a unique user name before allowing them to access system components or cardholder data"
```

File: pci_dss_v321/requirement_8.sp [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    benchmark.pci_dss_v321_requirement_8_1_2,
+    benchmark.pci_dss_v321_requirement_8_1_3,
+    benchmark.pci_dss_v321_requirement_8_1_5
+  ]
+
+  tags = local.pci_dss_v321_common_tags
+}
+
+benchmark "pci_dss_v321_requirement_8_1_2" {
+  title       = "Control addition, deletion, and modification of user IDs, credentials, and other identifier objects"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:25]:
```suggestion
benchmark "pci_dss_v321_requirement_8_1_2" {
  title = "Control addition, deletion, and modification of user IDs, credentials, and other identifier objects"
```

File: pci_dss_v321/requirement_8.sp [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    control.iam_external_user_with_write_permission
+  ]
+
+  tags = merge(local.pci_dss_v321_common_tags, {
+    service = "Azure/ActiveDirectory"
+  })
+}
+
+benchmark "pci_dss_v321_requirement_8_1_3" {
+  title       = "Immediately revoke access for any terminated users"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:40]:
```suggestion
benchmark "pci_dss_v321_requirement_8_1_3" {
  title = "Immediately revoke access for any terminated users"
```

File: pci_dss_v321/requirement_8.sp [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    control.iam_deprecated_account
+  ]
+
+  tags = merge(local.pci_dss_v321_common_tags, {
+    service = "Azure/ActiveDirectory"
+  })
+}
+
+benchmark "pci_dss_v321_requirement_8_1_5" {
+  title       = "Manage IDs used by thid parties to access, support, or maintain system components via remote access as follows: - Enabled only during the time period needed and disabled when not in use. - Monitored when in use"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:52]:
```suggestion
benchmark "pci_dss_v321_requirement_8_1_5" {
  title = "Manage IDs used by thid parties to access, support, or maintain system components via remote access as follows: - Enabled only during the time period needed and disabled when not in use. - Monitored when in use""
```

File: pci_dss_v321/requirement_8.sp [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    control.iam_external_user_with_write_permission
+  ]
+
+  tags = merge(local.pci_dss_v321_common_tags, {
+    service = "Azure/ActiveDirectory"
+  })
+}
+
+benchmark "pci_dss_v321_requirement_8_2" {
+  title       = "Employ at least one of these to authenticate all users: something you know, such as a password or passphrase; something you have, such as a token device or smart card; or something you are, such as a biometric. Use strong authentication methods and render all passwords/passphrases unreadable during transmission and storage using strong cryptography"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:67]:
```suggestion
benchmark "pci_dss_v321_requirement_8_2" {
  title = "Employ at least one of these to authenticate all users: something you know, such as a password or passphrase; something you have, such as a token device or smart card; or something you are, such as a biometric. Use strong authentication methods and render all passwords/passphrases unreadable during transmission and storage using strong cryptography"
```

File: pci_dss_v321/requirement_8.sp [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    control.compute_vm_guest_configuration_installed_windows,
+    control.compute_vm_guest_configuration_with_no_managed_identity,
+    control.compute_vm_guest_configuration_with_user_and_system_assigned_managed_identity,
+    control.compute_vm_max_password_age_70_days_windows,
+    control.compute_vm_min_password_length_14_windows,
+    control.compute_vm_restrict_previous_24_passwords_resuse_windows,
+  ]
+
+  tags = merge(local.pci_dss_v321_common_tags, {
+    service =  "Azure/Compute"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:88]:
```suggestion
    service = "Azure/Compute"
```

File: pci_dss_v321/requirement_8.sp [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+  })
+}
+
+benchmark "pci_dss_v321_requirement_8_2_5" {
+  title       = "Do not allow an individual to submit a new password/phrase that is the same as any of the last four passwords/phrases he or she has used"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:93]:
```suggestion
benchmark "pci_dss_v321_requirement_8_2_5" {
  title = "Do not allow an individual to submit a new password/phrase that is the same as any of the last four passwords/phrases he or she has used"
```

File: pci_dss_v321/requirement_8.sp [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    control.compute_vm_guest_configuration_installed_windows,
+    control.compute_vm_guest_configuration_with_no_managed_identity,
+    control.compute_vm_guest_configuration_with_user_and_system_assigned_managed_identity,
+    control.compute_vm_max_password_age_70_days_windows,
+    control.compute_vm_min_password_length_14_windows,
+    control.compute_vm_restrict_previous_24_passwords_resuse_windows
+  ]
+
+  tags = merge(local.pci_dss_v321_common_tags, {
+    service =  "Azure/Compute"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:104]:
```suggestion
    service = "Azure/Compute"
```

File: pci_dss_v321/requirement_8.sp [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+  })
+}
+
+benchmark "pci_dss_v321_requirement_8_3" {
+  title       = "Secure all individual non-console administrative access and all remote access to the cardholder data environment using multi-factor authentication. This requires at least two of the three authentication methods described in 8.2 are used for authentication. Using one factor twice (e.g. using two separate passwords) is not considered multi-factor authentication. This requirement applies to administrative personnel with non-console access to the CDE from within the entity's network, and all remote network access (including for users, administrators, and third-parties) originating from outside the entity's network"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:109]:
```suggestion
benchmark "pci_dss_v321_requirement_8_3" {
  title = "Secure all individual non-console administrative access and all remote access to the cardholder data environment using multi-factor authentication. This requires at least two of the three authentication methods described in 8.2 are used for authentication. Using one factor twice (e.g. using two separate passwords) is not considered multi-factor authentication. This requirement applies to administrative personnel with non-console access to the CDE from within the entity's network, and all remote network access (including for users, administrators, and third-parties) originating from outside the entity's network"
```

File: pci_dss_v321/requirement_8.sp [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+  children = [
+    benchmark.pci_dss_v321_requirement_8_3_1
+  ]
+
+  tags = local.pci_dss_v321_common_tags
+}
+
+benchmark "pci_dss_v321_requirement_8_3_1" {
+  title       = "Incorporate multi-factor authentication for all non-console access into the CDE for personnel with administrative access"
```
Comment (madhushreeray30) [inline pci_dss_v321/requirement_8.sp:118]:
```suggestion
benchmark "pci_dss_v321_requirement_8_3_1" {
  title = "Incorporate multi-factor authentication for all non-console access into the CDE for personnel with administrative access"
```

**Review by madhushreeray30 — CHANGES_REQUESTED** [review]
@khushboo9024 can you please look into the following changes. Thanks!

**Review by madhushreeray30 — APPROVED** [review]

File: pci_dss_v321/requirement_6.sp [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+benchmark "pci_dss_v321_requirement_6_2" {
+  title       = "Protect all system components and software from known vulnerabilities by installing applicable vendor-supplied security patches"
+  description = "Install critical security patches within one month of release."
+  children = [
+    control.compute_vm_vulnerability_assessment_solution_enabled,
+    control.compute_vm_endpoint_protection_agent_installed,
+    control.sql_database_vulnerability_findings_resolved,
+    control.compute_vm_system_updates_installed,
+    control.compute_vm_security_configuration_vulnerabilities_remediated
```
Comment (misraved) [inline pci_dss_v321/requirement_6.sp:21]: Please sort the controls.

File: pci_dss_v321/requirement_7.sp [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+
+benchmark "pci_dss_v321_requirement_7_1_1" {
+  title = "Define access needs for each role, including: - System components and data resources that each role needs to access for their job function - Level of privilege required (for example, user, administrator, etc.) for accessing resources"
+  children = [
+    control.iam_subscription_owner_max_3,
+    control.iam_subscription_owner_more_than_1
+  ]
+
+  tags = merge(local.pci_dss_v321_common_tags, {
+    service = "Azure/ActiveDirectory"
```
Comment (misraved) [inline pci_dss_v321/requirement_7.sp:31]: Should the service be `Azure/IAM`?

Should all the benchmarks refer to the `Azure/IAM` service?

File: pci_dss_v321/requirement_7.sp [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+benchmark "pci_dss_v321_requirement_7_2_1" {
+  title = "Coverage of all system components"
+  children = [
+    control.sql_server_azure_ad_authentication_enabled,
+    control.iam_no_custom_role,
+    control.iam_external_user_with_owner_role,
+    control.iam_external_user_with_read_permission,
+    control.iam_external_user_with_write_permission,
+    control.iam_user_with_write_permission_on_subscription_mfa_enabled,
+    control.iam_user_with_owner_permission_on_subscription_mfa_enabled
```
Comment (misraved) [inline pci_dss_v321/requirement_7.sp:77]: Please sort the controls.

File: pci_dss_v321/requirement_8.sp [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+benchmark "pci_dss_v321_requirement_8_2_3" {
+  title       = "Passwords/phrases must meet the following: - Require a minimum length of at least seven characters. - Contain both numeric and alphabetic characters"
+  description = "Alternatively, the passwords/phrases must have complexity and strength at least equivalent to the parameters specified above."
+  children = [
+    control.compute_vm_guest_configuration_installed_windows,
+    control.compute_vm_guest_configuration_with_no_managed_identity,
+    control.compute_vm_guest_configuration_with_user_and_system_assigned_managed_identity,
+    control.compute_vm_max_password_age_70_days_windows,
+    control.compute_vm_min_password_length_14_windows,
+    control.compute_vm_restrict_previous_24_passwords_resuse_windows,
```
Comment (misraved) [inline pci_dss_v321/requirement_8.sp:88]:
```suggestion
    control.compute_vm_restrict_previous_24_passwords_resuse_windows
```

File: pci_dss_v321/requirement_8.sp [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+benchmark "pci_dss_v321_requirement_8_3_1" {
+  title = "Incorporate multi-factor authentication for all non-console access into the CDE for personnel with administrative access"
+  children = [
+    control.iam_external_user_with_owner_role,
+    control.iam_external_user_with_read_permission,
+    control.iam_external_user_with_write_permission,
+    control.iam_no_custom_role,
+    control.iam_user_with_owner_permission_on_subscription_mfa_enabled,
+    control.iam_user_with_write_permission_on_subscription_mfa_enabled,
+    control.sql_server_azure_ad_authentication_enabled,
```
Comment (misraved) [inline pci_dss_v321/requirement_8.sp:131]:
```suggestion
    control.sql_server_azure_ad_authentication_enabled
```

**Review by misraved — COMMENTED** [review]
@khushboo9024 please take a look at the review comments. 

Additionally please make sure that all the controls inside a benchmark are sorted, and also update the mod.sp file to include PCI information.

File: pci_dss_v321/requirement_7.sp (hunk 16 above)
Comment (khushboo9024) [reply pci_dss_v321/requirement_7.sp:31]: Service should be `Azure/ActiveDirectory`

**Pull request merged**